plugin/pulpcore/plugin/stages/declarative_version.py

```diff
@@ -82,14 +81,8 @@ def __init__(self, first_stage, repository, sync_mode='mirror'):
                  :class:`~pulpcore.plugin.stages.DeclarativeVersion stream`, and does not remove any
                  pre-existing units in the :class:`~pulpcore.plugin.models.RepositoryVersion`.
                  'mirror' is the default.
-
-        Raises:
-            ValueError: if 'sync_mode' is passed an invalid value.
         """
-
-        if sync_mode != 'mirror' and sync_mode != 'additive':
-            msg = _("'sync_mode' must either be 'mirror' or 'additive' not '{sync_mode}'")
-            raise ValueError(msg.format(sync_mode=sync_mode))
+        assert(sync_mode == 'mirror' or sync_mode == 'additive')
```

Comment:
I'm not sure I understand this change?

Reply:
Ah, I see now. The value is being validated by the serializer.

Reply:
I think a ValueError is more correct than an AssertionError here. Look at the description of ValueError in the standard library docs. https://docs.python.org/3/library/exceptions.html#ValueError

Reply:
This was based on my suggestion so I'm to blame ;) The case will be more obvious if proper constants and namespacing are used:

```python
class DeclarativeVersion:
    sync_mode_mirror = 'mirror'
    sync_mode_additive = 'additive'
    default_sync_mode = sync_mode_mirror

    def __init__(self, first_stage, repository, sync_mode=None):
         ...

        if sync_mode is None:
           sync_mode = self.default_sync_mode
       assert sync_mode in {self.sync_mode_additive, sync_mode_mirror}, 'Oops!'
```
 IMO a declarative_version object always expects just the values/switches it was declared with therefore the `assert` statement is just a debugging thing.